src/diffpy/utils/scattering_objects/diffraction_objects.py

```diff
@@ -21,6 +21,17 @@ def __init__(self, name='', wavelength=None):
         self._all_arrays = [self.on_q, self.on_tth]
         self.metadata = {}
 
+    def __eq__(self, other):
+        if not isinstance(other, Diffraction_object):
+            return NotImplemented
+        return (self.name == other.name and
```

Comment:
can we think of a way of doing this by iterating over `DO.__dict__.items()`?  this way it will automatically update in the future when we add attributes.

Reply:
Oh this makes sense, I'll think about it

Reply:
For datetime, I think we can set `self.creation_time = datetime.now()` (we can convert it into a string if this makes everything easier), and skip certain attributes (use `reserved_keys = [creation_time]`), and I think everything else works basically the same